service/controller/resource/instance/template/main.json

```diff
@@ -158,10 +150,219 @@
       "type":"Microsoft.Resources/deployments",
       "name":"worker-vmss-deploy",
       "properties":{
+        "expressionEvaluationOptions":{
+          "scope": "inner"
+        },
```

Comment:
This is required when we want references (parameters or variables) in the nested template to reference variables/parameters defined in the template. Otherwise it will reference variables/parameters from parent template.

If this works fine, we can later see if we can remove some parameters from nested templates.